java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -19,15 +19,41 @@
  */
 package org.sonar.java.se;
 
+import org.sonar.java.collections.PMap;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.semantic.Type;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
 import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
+import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
 
 public class ConstraintManager {
 
+  private static final String CLOSE_METHOD_NAME = "close";
+  private static final String JAVA_IO_CLOSEABLE = "java.io.Closeable";
+
+  private static final String[] IGNORED_CLOSEABLE_SUBTYPES = {
+    "java.io.ByteArrayOutputStream",
+    "java.io.ByteArrayInputStream",
+    "java.io.StringReader",
+    "java.io.StringWriter",
+    "java.io.CharArrayReader",
+    "java.io.CharArrayWriter",
+    "com.sun.org.apache.xml.internal.security.utils.UnsyncByteArrayOutputStream"
+  };
+
   private int counter = ProgramState.EMPTY_STATE.constraintsSize();
 
+  public SymbolicValue createWrappedSymbolicValue(SymbolicValue dependent) {
+    return new SymbolicValue.ResourceWrapperSymbolicValue(counter++, dependent);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this increment or decrement operator into a dedicated statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS881)